flink-runtime/src/main/java/org/apache/flink/runtime/metrics/DescriptiveStatisticsHistogram.java

```diff
@@ -22,25 +22,30 @@
 
 import org.apache.commons.math3.stat.descriptive.DescriptiveStatistics;
 
+import java.util.concurrent.atomic.AtomicLong;
+
 /**
  * The {@link DescriptiveStatisticsHistogram} use a DescriptiveStatistics {@link DescriptiveStatistics} as a Flink {@link Histogram}.
  */
 public class DescriptiveStatisticsHistogram implements org.apache.flink.metrics.Histogram {
 
 	private final DescriptiveStatistics descriptiveStatistics;
 
+	private final AtomicLong elementsSeen = new AtomicLong();
```

Comment:
Is this class assumed to be use in a multi-threaded context?

Reply:
that depends on how users use it - usually, there should be >=1 writer thread(s) (tasks) and 1 reader (the metrics reporter)

Reply:
All metrics are read in a multi-thread context; the reporter thread and MetricQueryService can run concurrently.

Reply:
I would not try to handle concurrent write access btw. (users are responsible for ensuring single-threaded access)

Reply:
Seems like the underlying code is not thread-safe anyways and we provide only best-effort metrics. I reverted to just using a `long` now.